src/Quasar/FS/Resource.purs

```diff
@@ -39,21 +40,21 @@ instance showQResource ∷ Show QResource where
   show (Directory p) = "(Directory " <> show p <> ")"
   show (Mount m) = "(Mount " <> show m <> ")"
 
-fromJSON ∷ DirPath → Json → Either String QResource
+fromJSON ∷ AbsDir → Json → Either String QResource
 fromJSON parent json
   = Mount <$> Mount.fromJSON parent json
   <|> do
     obj ← decodeJson json
-    name ← obj .? "name"
+    name' ← note "empty name" <<< fromString =<< (obj .? "name")
```

Comment:
I'm not sure about this, since there can be a mount at `/` (which will have no name, understandably). I'll try and concoct a test case.

Reply:
This is fine actually, it's just for file listings, so everything must indeed have a name.